### PR TITLE
Ere 398

### PR DESCRIPTION
**Pull request by barelyThinkingBagOfWater**
Practitioner information now filled from extraction to bundle, compatible with DENS pdf1 + both pdf from CGM. Tests of parser + ExtractionToBundleWorkflowTest completed.

Also fixed some fields (including birthdate and authoredOn) in RegexParser, last time I fixed them in Muster16SvgExtractorParser as I didn’t know the regexParser existed. Javadoc added in the extractorParser to indicate it’s deprecated.

I’ve attached in the Jira task the test1.pdf with the number in the practitioner name removed as we shouldn’t support number in this field (or should we?) and as I couldn’t push in the secret repo, just add it in the same folder as test1.pdf.

### CODE REVIEW

**Comment by cartel1** [comment]
This looks OK to me.

@OmarMalass please confirm so we can merge this.  Thanks.

File: src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/pattern/PractitionerPatterns.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package health.ere.ps.service.muster16.parser.rgxer.delegate.pattern;
+
+import java.util.regex.Pattern;
+
+public class PractitionerPatterns extends Patterns {
+    public final Pattern NAME_PREFIX = Pattern.compile("(Dr)\\.");
```
Comment (OmarMalass) [inline src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/pattern/PractitionerPatterns.java:6]: Other possible prefixes: (med. prof.), can be a composite prefix like: "Dr. med."

File: src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/pattern/PractitionerPatterns.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    public final Pattern NAME_LINE = Pattern.compile("(Dr\\.)*([a-z A-Z]+)(-){0,1}([a-z A-Z]+)");
+    public final Pattern STREET_LINE = Pattern.compile("^[a-z A-Zß]+(\\.{1})?.*[0-9]+");
```
Comment (OmarMalass) [inline src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/pattern/PractitionerPatterns.java:8]: There are German characters that won't be captured by the letter pattern. You can use \\w instead.

File: src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/practitioner/PractitionerEntryParseDelegate.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        List<String> lines = Arrays.stream(entry.split("\\n"))
+                .map(String::trim)
+                .collect(Collectors.toList());
+
+        if (lines.size() >= 5) {
+            matchAndExtractLine(lines, patterns.FAX_LINE).ifPresent(this::parseFaxNumber);
+            matchAndExtractLine(lines, patterns.PHONE_LINE).ifPresent(this::parsePhoneNumber);
+            matchAndExtractLine(lines, patterns.CITY_LINE).ifPresent(this::parseAddressLine);
+            matchAndExtractLine(lines, patterns.STREET_LINE).ifPresent(this::parseStreetLine);
+            matchAndExtractLine(lines, patterns.NAME_LINE).ifPresent(this::parseNames);
```
[the remark this answers is not in the document]
Comment (barelyThinkingBagOfWater) [reply src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/practitioner/PractitionerEntryParseDelegate.java:40]: You mean the type of doctor he is, like dentist or general doctor? I would say here let's wait until we need to do it to have a clearer idea of what we need to support. I've heard that dentists will be one of our main user soon,

File: src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/pattern/PractitionerPatterns.java (hunk 2 above)
Comment (barelyThinkingBagOfWater) [reply src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/pattern/PractitionerPatterns.java:8]: \w actually matches [A-Za-z0-9_], not any German character. I've added äöüÄÖÜß to the list of accepted letters, that will do the trick :) 
I don't think we should match whitespace here as such character would involve an error or weird behavior in our parsing.

File: src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/pattern/PractitionerPatterns.java (hunk 1 above)
Comment (barelyThinkingBagOfWater) [reply src/main/java/health/ere/ps/service/muster16/parser/rgxer/delegate/pattern/PractitionerPatterns.java]: Ok, support of them added.

**Pull request merged**